src/arch/x86/smbios.c

```diff
@@ -594,6 +594,14 @@ static int smbios_write_type11(unsigned long *current, int *handle)
 	return len;
 }
 
+static int smbios_write_type16(unsigned long *current, int *handle)
+{
+	int len = fill_mainboard_smbios_type16(current, handle);
+	*current += len;
+	(*handle)++;
```

Comment:
@finus16 if `len` is 0 then no type16 has been added. The `handle` pointer should remain unchanged in such case. Don't You think?

Reply:
fixed